opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1362,73 +1362,117 @@ template cpgrid::Entity<3> Dune::createEntity(const CpGrid&, int, bool);
+        const auto& [level_ptr, boundary_old_to_new_corners, boundary_old_to_new_faces, parent_to_children_faces,
+                     parent_to_children_cells, child_to_parent_faces, child_to_parent_cells]
+            = (*(this-> data_[0])).refinePatch(cells_per_dim_vec[patch], startIJK_vec[patch], endIJK_vec[patch]);
+        // Add each LGR to "data".
+        (this-> data_).push_back(level_ptr);
+        (*data_[patch +1]).level_data_ptr_ = &(this -> data_); // shifted +1 since level0 is coarse grid. Levels are 1,2,..., num_patches.
+        (*data_[patch +1]).level_ = patch +1;
+        //
+        // POPULATING (*data_[0]).parent_to_children_cells_
+        // POPULATING (*data_[patch +1]).child_to_parent_cells_
+        // Extension of container all_patch_child_to_parent_cells, adding {-1,-1} entries for each NO-CHILD cell, in each level/patch.
+        (all_patch_child_to_parent_cells[patch]).resize(child_to_parent_cells.size()); // To consider all cells in the LGR.
+        // True parent cells have entries: {level of the LGR the parent cell has its children, {child0, child1, ...}}
+        // False parent cells (NO PARENT CELLS) have {-1,{-1}} entries.
+        // parent_to_children_cells entries look like {parent cell index (in level 0), {child0, child1,..}}
+        // True child cells have entries: {0, parent cell index} (0 represents the "GLOBAL" coarse grid)
+        // False child cells (with no parent) have {-1,-1} entries.
+        // child_to_parent_cells entries look like {child index in the LGR, parent cell index} 
+        // Re-write entries of actual parent/Create the ones for child cells in each level (not default value {-1,-1} needed).
+        assert(!parent_to_children_cells.empty());
+        for (const auto& [trueParent, children_list] : parent_to_children_cells){
+            l0_parent_to_children_cells[trueParent] = std::make_tuple(patch +1, children_list); // {level/LGR, {child0, child1, ...}}
+            // shifted +1 since "GLOBAL" is level0 (coarse grid). Levels are 1,2,..., num_patches.
+            assert(!children_list.empty());
+            for (const auto& child : children_list){
+                all_patch_child_to_parent_cells[patch][child] = {0, trueParent};
+                // {level of parent cell, parent cell index in that level}
+            }
+        }
+        // Add this family information [child_to_parent_cells_] in each CpGridData object representing each LGR
+        (*data_[patch +1]).child_to_parent_cells_ =  all_patch_child_to_parent_cells[patch];
+        // shifted +1 since "GLOBAL" is level0 (coarse grid). Levels are 1,2,..., num_patches.
+        //
+        all_patch_boundary_old_to_new_corners.push_back(boundary_old_to_new_corners);
+        all_patch_boundary_old_to_new_faces.push_back(boundary_old_to_new_faces);
+    } // end-patch-forloop
+    // LEVEL 0, definition/declaration of some members:
+    (*data_[0]).level_data_ptr_ = &(this -> data_);
+    (*data_[0]).level_ = 0;
```

Comment:
I would hope that this not needed and already correctly initialized previously. Isn't that the case?

Reply:
level_data_ptr_ was already correctly initialized previously. It wasn't the case for level_. Now both are correctly initialized so those lines are gone in the current code.